crates/astria-composer/src/searcher/executor.rs

```diff
@@ -0,0 +1,333 @@
+use std::time::Duration;
+
+use color_eyre::eyre::{
+    self,
+    bail,
+    eyre,
+    Context,
+};
+use ed25519_consensus::SigningKey;
+use humantime::format_duration;
+use proto::native::sequencer::v1alpha1::{
+    Action,
+    SignedTransaction,
+    UnsignedTransaction,
+};
+use secrecy::{
+    ExposeSecret as _,
+    SecretString,
+    Zeroize as _,
+};
+use sequencer_client::{
+    Address,
+    NonceResponse,
+    SequencerClientExt,
+};
+use tokio::sync::{
+    mpsc,
+    watch,
+};
+use tracing::{
+    debug,
+    error,
+    info,
+    instrument,
+    warn,
+};
+
+use crate::Config;
+
+pub(super) type StatusReceiver = watch::Receiver<Status>;
+pub(super) type Sender = mpsc::Sender<Vec<Action>>;
+
+pub(super) fn spawn(cfg: &Config) -> eyre::Result<(Sender, StatusReceiver)> {
+    info!("Spawning Executor subtask for Searcher");
+    // create channel for sending bundles to executor
+    let (executor_tx, executor_rx) = mpsc::channel(256);
+    let executor = Executor::new(&cfg.sequencer_url, &cfg.private_key, executor_rx)?;
+
+    // create channel for receiving executor status
+    let status_rx = executor.subscribe();
+
+    // spawn executor task
+    let join_handle = tokio::spawn(executor.run_until_stopped());
+
+    // handle executor failure by logging
+    tokio::task::spawn(async move {
+        match join_handle.await {
+            Ok(Ok(())) => {
+                error!("executor task exited unexpectedly");
+            }
+            Ok(Err(e)) => {
+                error!(
+                    error.message = %e,
+                    error.cause_chain = ?e,
+                    "executor task failed unexpectedly with error",
+                );
+            }
+            Err(e) => {
```

Comment:
I feel like this case and the case above can be combined. If there is meaningful distinction between the two, they don't seemed to be clear from how the branches are handled

Reply:
They are for handling different errors. This one is for the executor task panicking and the other is for it returning an error - since they're different code paths I feel like they should be reported differently.